File: backend/app/schemas/help_chat.py
from __future__ import annotations

from pydantic import BaseModel, Field


class HelpChatMessage(BaseModel):
    role: str = Field(..., pattern="^(user|assistant)$")
    content: str


class HelpHighlightTarget(BaseModel):
    help_id: str
    label: str
    reason: str


class HelpChatLink(BaseModel):
    href: str
    label: str


class HelpChatRequest(BaseModel):
    message: str
    pathname: str = "/"
    search: str | None = None
    page_title: str | None = None
    page_description: str | None = None
    visible_help_ids: list[str] = Field(default_factory=list)
    history: list[HelpChatMessage] = Field(default_factory=list)


class HelpChatResponse(BaseModel):
    reply: str
    highlights: list[HelpHighlightTarget] = Field(default_factory=list)
    links: list[HelpChatLink] = Field(default_factory=list)
    ai_enhanced: bool = False
    page_label: str | None = None

File: frontend/src/lib/help-chat.ts
export interface HelpHighlight {
  helpId: string;
  label: string;
  reason: string;
}

export interface HelpChatLink {
  href: string;
  label: string;
}

export interface HelpChatMessage {
  id: string;
  role: "user" | "assistant";
  content: string;
  highlights?: HelpHighlight[];
  links?: HelpChatLink[];
  aiEnhanced?: boolean;
}

export function collectVisibleHelpIds(): string[] {
  if (typeof document === "undefined") return [];
  return Array.from(document.querySelectorAll("[data-help-id]"))
    .map((el) => el.getAttribute("data-help-id"))
    .filter((id): id is string => Boolean(id));
}

export function highlightHelpTarget(helpId: string): boolean {
  const el = document.querySelector(`[data-help-id="${helpId}"]`);
  if (!el || !(el instanceof HTMLElement)) return false;

  el.scrollIntoView({ behavior: "smooth", block: "center", inline: "nearest" });
  el.classList.add("help-spotlight");
  window.setTimeout(() => el.classList.remove("help-spotlight"), 3200);
  return true;
}

export function clearHelpSpotlights() {
  document.querySelectorAll(".help-spotlight").forEach((el) => el.classList.remove("help-spotlight"));
}
